test/programs/DiagrammaticPrograms.jl

```diff
@@ -577,4 +545,323 @@ F_src_vv, F_src_ev, F_src_ve = components(diagram_map(F_src))
 @test get_keyword_arg_val(:(x=3)) == 3
 @test_throws ErrorException get_keyword_arg_val(:("not an assignment!"+3))
 @test destructure_unary_call(:(f(g(x)))) == (:(f∘g),:x)
+
+# Migrations with code
+#
+#------------------------------------
+
+
+
+
+#@testset "Migrations with code" begin
+  @present SchMechLink <: SchGraph begin
+      Pos::AttrType
+      Len::AttrType
+      pos::Attr(V,Pos)
+      len::Attr(E,Len)
+  end
+  @acset_type MechLink(SchMechLink, index=[:src,:tgt])
+  G = @acset MechLink{Vector{Float64},Float64} begin
+      V = 3
+      E = 2
+      src = [1,2]
+      tgt = [2,3]
+      len = [1.0,1.0]
+      pos = [[1.0,1.0,1.0],[2.0,2.0,2.0],[2.0,2.0,1.0]]
+  end
+  
+  #Rotate the whole linkage by a bit
+  M = @migration SchMechLink SchMechLink begin
+      V => V
+      E => E
+      Pos => Pos
+      Len => Len
+      src => src
+      tgt => tgt
+      pos => begin 
+              θ = π/5
+              M = [[cos(θ),sin(θ),0] [-sin(θ),cos(θ),0] [0,0,1]]
+              x -> M*pos(x)
+              end
+      len => len
+  end
+  A = migrate(MechLink,G,M)
```

Comment:
`migrate` should be tested in `categorical_algebra/DataMigrations.jl` rather than here.